File: EcmApiRestService/RestService/Models/FolderInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Agilent.ECMAPI;

namespace EcmapiRestService.Models
{
    public class FolderInfo
    {
        public string Name;
        public string ID;
        public string Path;
        public string ParentID;
        public int Level;
        public List<EcmapiRestService.Models.FolderInfo> folderinfos = new List<EcmapiRestService.Models.FolderInfo>();
        public List<EcmapiRestService.Models.FileInfo> fileinfos = new List<EcmapiRestService.Models.FileInfo>();

        public FolderInfo() { }

        public FolderInfo(ECM ecm, Agilent.ECMAPI.FolderInfo folderInfo, bool includeFiles)
            : this(ecm, folderInfo)
        {
            if (includeFiles)
            {
                Agilent.ECMAPI.FileInfos fileInfos = folderInfo.GetFileInfos();
                foreach (Agilent.ECMAPI.FileInfo fileInfo in fileInfos)
                {
                    fileinfos.Add(new EcmapiRestService.Models.FileInfo(fileInfo));
                }
            }
        }

        public FolderInfo(ECM ecm, Agilent.ECMAPI.FolderInfo folderInfo)
        {
            this.Name = folderInfo.Name;
            this.ID = folderInfo.ID.ToString();
            this.Path = folderInfo.Path;
            this.Path = ecm.GetFullPathFromID(folderInfo.ID);
            this.ParentID = folderInfo.ParentID.ToString();
            this.Level = folderInfo.GetFolderLevel();
            foreach (Agilent.ECMAPI.FolderInfo childFolderInfo in folderInfo.GetFolderInfos())
            {
                folderinfos.Add(new EcmapiRestService.Models.FolderInfo(ecm, childFolderInfo));
            }
        }
    }
}

File: EcmApiClient/src/com/pfizer/ecmapi/client/models/FolderInfo.java
package com.pfizer.ecmapi.client.models;

import java.util.ArrayList;
import java.util.List;

import com.thoughtworks.xstream.annotations.XStreamAlias;

@XStreamAlias("FolderInfo")
public class FolderInfo {
	@XStreamAlias("ID")
    private String id;
	@XStreamAlias("Name")
	private String name;
	@XStreamAlias("ParentID")
	private String parentID;
	@XStreamAlias("Path")
	private String path;
	private List<FolderInfo> folderinfos = new ArrayList<FolderInfo>();
	private List<FileInfo> fileinfos = new ArrayList<FileInfo>();
	@XStreamAlias("Level")
    private int level;

    public FolderInfo() { }

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getPath() {
		return path;
	}

	public void setPath(String path) {
		this.path = path;
	}

	public String getParentID() {
		return parentID;
	}

	public void setParentID(String parentID) {
		this.parentID = parentID;
	}

	public int getLevel() {
		return level;
	}

	public void setLevel(int level) {
		this.level = level;
	}

	public List<FolderInfo> getFolderinfos() {
		return folderinfos;
	}

	public void setFolderinfos(List<FolderInfo> folderinfos) {
		this.folderinfos = folderinfos;
	}

	public List<FileInfo> getFileinfos() {
		return fileinfos;
	}

	public void setFileinfos(List<FileInfo> fileinfos) {
		this.fileinfos = fileinfos;
	}

}
